File: 283-move-zeroes/move-zeroes.js
/**
 * @param {number[]} nums
 * @return {void} Do not return anything, modify nums in-place instead.
 */
var moveZeroes = function(a) {
    let x = 0;
    for(let i=0;i<a.length;i++) {
        if(a[i]!==0){
            a[x] = a[i];
            x++
        }
    }
    //fill all left-overs with zeroes
    for( let i=x;i<a.length;i++) {
        a[i] = 0;
    }
    return a;
};

File: 283-move-zeroes/move-zeroes.java
class Solution {
    public void moveZeroes(int[] arr) {
        
        int x = 0;

        for(int i=0;i<arr.length;i++) {

            if(arr[i]!=0) {
                arr[x] = arr[i];
                x++;

            }
        }

        for(int i=x;i<arr.length;i++) {
            arr[i]=0;
        }

    }
}